README.md

```diff
@@ -1,9 +1,236 @@
+
+```toml
+[dependencies]
+cdk_framework = { git = "https://github.com/demergent-labs/cdk_framework", rev = "08a74a4077d0981acde2b0fc714e877bba659b9e" }
+quote = "1.0.21"
+proc-macro = "1.0.43"
+```
+
+The CDK framework will help to generate the lib.rs file for you canister, but you cdk will still need to provide everything else.
+
+### Abstract Canister Tree
+
+To generate the rust token_stream for the lib.rs you simply need to construct an AbstractCanisterTree
+
+For example
+
+```rust
+fn main() {
+    // TODO generate AST representation of the canister you are parsing
+
+    let cdk_name = "YOUR_CDK_NAME".to_string() // For example: "kybra".to_string()
+
+    let header; // TODO generate any rust code your cdk needs to come at the beginning of the generated lib file
+    let body;   // TODO generate any rust code your cdk needs to have in the body of the generated lib file
+
+    let heartbeat_method;       // TODO build heartbeat method,
+    let init_method;            // TODO build init method,
+    let inspect_message_method; // TODO build inspect method,
+    let post_upgrade_method;    // TODO build post upgrade method,
+    let pre_upgrade_method;     // TODO build pre upgrade method,
+    let query_methods;          // TODO build query methods,
+    let update_methods;         // TODO build update methods,
+
+    let guard_functions;    // TODO build guard functions
+
+    let funcs;          // TODO build funcs,
+    let records;        // TODO build records,
+    let tuples;         // TODO build tuples,
+    let type_aliases;   // TODO build type aliases,
+    let variants;       // TODO build variants,
+
+    let try_from_vm_value_impls; // TODO generate try into vm value impls,
+    let try_into_vm_value_impls; // TODO generate try from vm value impls,
+
+    let keywords;   // TODO generate a list of keyword from your CDK's language
+    // For example vec!["for", "if", "int", "import", "bool", "while", etc...]
+
+    let canister_methods = CanisterMethods {
+        heartbeat_method,
+        init_method,
+        inspect_message_method,
+        post_upgrade_method,
+        pre_upgrade_method,
+        query_methods,
+        update_methods,
+    };
+
+    let candid_types = CandidTypes {
+        funcs,
+        records,
+        tuples,
+        type_aliases,
+        variants,
+    };
+
+    let vm_value_conversion = VmValueConversion {
+        try_from_vm_value_impls,
+        try_into_vm_value_impls,
+    };
+
+    let lib_file = AbstractCanisterTree {
+        cdk_name,
+        header,
+        body,
+        candid_types,
+        canister_methods,
+        external_canisters,
+        guard_functions,
+        vm_value_conversion,
+        keywords,
+    }.to_token_stream().to_string()
+
+    // TODO write contents of lib_file to a file
+
+    // TODO generate cargo.toml
+
+    // TODO build your generated rust project and deploy to replica
+}
+
+```
+
+### Candid Types
```

Comment:
Match the order, and have a little bit better of a transition saying that we are now explaining each for the fields. Give a small table of contents